scripts/image-build/build-vyos-image

```diff
@@ -607,15 +607,17 @@ if __name__ == "__main__":
     if build_config["image_format"] != ["iso"]:
         raw_image = raw_image.create_raw_image(build_config, iso_file, "tmp/")
 
-        other_formats = filter(lambda x: x not in ["iso", "raw"], build_config["image_format"])
-        for f in other_formats:
-            target = f"{os.path.splitext(raw_image)[0]}.{f}"
-            print(f"I: building {f} file {target}")
-            os.system(f"qemu-img convert -f raw -O {f} {raw_image} {target}")
-
-        # Some flavors require special procedures that aren't covered by qemu-img
-        # (most notable, the VMware OVA that requires a custom tool to make and sign the image).
-        # Such procedures are executed as post-build hooks.
         if has_nonempty_key(build_config, "post_build_hook"):
+            # Some flavors require special procedures that aren't covered by qemu-img
+            # (most notably, the VMware OVA that requires a custom tool to make and sign the image).
+            # For those cases, we support running a post-build hook on the raw image.
+            # The image_format field should be 'raw' if a post-build hook is used.
             hook_path = build_config["post_build_hook"]
             os.system(f"{hook_path} {raw_image}")
+        else:
+            # Most other formats, thankfully, can be produced with just `qemu-img convert`
+            other_formats = filter(lambda x: x not in ["iso", "raw"], build_config["image_format"])
+            for f in other_formats:
+                target = f"{os.path.splitext(raw_image)[0]}.{f}"
+                print(f"I: Building {f} file {target}")
+                os.system(f"qemu-img convert -f raw -O {f} {raw_image} {target}")
```

Comment:
Should this be try/catch wrapped? In event conversion is for some reason not possible or fails?

Reply:
Fair point! I'll add it.

Reply:
Is it better to use `subprocess.run` instead of the old `os.system`?

Reply:
It now uses `vyos.subprocess.call` and the script actually fails on command execution errors now (`os.system` doesn't raise exceptions). I think it's probably better to just make it raise exceptions in all cases since it's a developer tool and those commands are normally expected to work for valid inputs.